File: Offer2/0010.2/main.go
package main

func numWays(n int) int {
	if n <= 1 {
		return 1
	}
	if n == 2 {
		return 2
	}
	const mod int = 1e9 + 7
	v1, v2 := 1, 2
	var i int
	for i = 2; i < n; i++ {
		v1, v2 = v2, (v1+v2)%mod
	}
	return v2
}

File: Offer2/0010.2/main.py
class Solution:
    def numWays(self, n: int) -> int:
        if n <= 1:
            return 1
        if n == 2:
            return 2
        
        mod: int = 1e9 + 7
        v1, v2 = 1, 2
        for _ in range(2, n):
            v1, v2 = v2, (v1 + v2) % mod
        return int(v2)